### DISCUSSION TITLE
WIP - External stand-alone Devtool

### DISCUSSION THREAD

**Discussion by Thiesl137** [pmndrs/jotai#918] [Show and tell]
Hello Everyone! 

My team and I have developed a minimal stand-alone devtool Atomic for React applications using Jotai! 
It's a WIP and have just released a beta version for public use. 

You can visit the [Website](https://atomictool.dev/) or [GitHub](https://github.com/oslabs-beta/atomic) to get more information.

It's currently under review with Chrome Web Store so not accessible by download yet. In the mean-time, if you would like to explore the tool you can fork the repository and load unpacked the extension/build folder into your chrome extensions.

Key features of Atomic Devtool include:

- Display state changes between snapshots
- Display component tree structure
- Dynamically render graphic component visualization with state changes
- Atom network data visualization for both atom dependents and atom readDependencies

It's an open-source library and welcome contributions and feedback!

As soon as internal snapshot API @Thisen is working on #425 is finished, we plan to integrate Atomic with Jotai's snapshot API. 

Thanks!

**Comment by Thiesl137** [comment]
The devtool is now available for download on the Chrome Web Store [here](https://chrome.google.com/webstore/detail/atomic-dev-tool/nnchkolpjdpkpbolophmdkglenapodbo)! 

Again, publish issues and feedback so we can fine tune for snapshots API.

**Comment by dai-shi** [comment]
@Thiesl137 #474 is merged, and it's basically ready. what's remaining is docs, but you can play around with it.

**Comment by dai-shi** [comment]
Docs PR: https://github.com/pmndrs/website/pull/87

**Comment by Thisen** [comment]
Hi @Thiesl137, what's the status on this?

**Comment by Thiesl137** [comment]
Hey @Thisen, It's been a journey - as well, this summer was busier than I had anticipated. None the less, I have been able to figure out how to implement the snapshot API to enable time-travel with the devtool on a basic Jotai application. I'll spend tomorrow cleaning up the package and reloading to the Chrome Store.

**Comment by dai-shi** [comment]
Any updates?

**Comment by dai-shi** [comment]
I opened a new issue #911. I think there's room for improvement.
It would be so nice to get some feedbacks from Atomic Devtool team.
